src/starspace.cpp

```diff
@@ -342,9 +343,21 @@ Metrics StarSpace::evaluateOne(
   // get the first K predictions
   int i = 0;
   while (i < args_->K && heap.size() > 0) {
-    pred.push_back(heap.top());
+    Predictions heap_top = heap.top();
     heap.pop();
-    i++;
+
+    bool keep = true;
+    if(excludeLHS) {
+      int nwords = dict_->nwords();
+      auto it = std::find_if( lhs.begin(), lhs.end(),
+                             [&heap_top, &nwords](const Base& el){ return (el.first - nwords + 1) == heap_top.second;} );
```

Comment:
I think it should be (el.first - nwords) == heap_top.second here.
In addition, keep in mind that one can also supply a baseDoc file and the elements in baseDoc can be in any order. Could you make it also work if -baseDoc is provided in eval?

There is also the case where -fileFormat=labelDoc. In that case I think we can skip this check.

Reply:
I think `el.first - nwords + 1` is correct, since it's `i + 1` on [L340](https://github.com/bkj/StarSpace/blob/6351d26dd056dd8f0e6007df49b4c42cc09f9d5a/src/starspace.cpp#L340) .  

(This way worked in a test as well, while `el.first - nwords` did not.)